x/stakeibc/keeper/hooks.go

```diff
@@ -53,20 +54,56 @@ func (k Keeper) BeforeEpochStart(ctx sdk.Context, epochIdentifier string, epochN
 				}
 			}
 		}
+
+		// DELEGATE FROM DELEGATION ACCOUNT
 		delegateInterval := int64(k.GetParam(ctx, types.KeyDelegateInterval))
 		if epochNumber%delegateInterval == 0 {
+			// get Gaia LC height
 			k.ProcessDelegationStaking(ctx)
 		}
+
 		exchangeRateInterval := int64(k.GetParam(ctx, types.KeyExchangeRateInterval))
-		if epochNumber%exchangeRateInterval == 3 { // allow a few blocks from UpdateUndelegatedBal to avoid conflicts
-			// TODO(TEST-97) update only when balances, delegatedBalances and stAsset supply are results from the same block
-			k.ProcessUpdateDelegatedBalance(ctx)
-		}
-		if epochNumber%exchangeRateInterval == 6 && (epochNumber > 100) { // allow a few blocks from UpdateDelegatedBal to avoid conflicts & wait until chain has registered zones to calc exch rate
-			// TODO(TEST-97) update only when balances, delegatedBalances and stAsset supply are results from the same block
-			k.ProcessUpdateExchangeRate(ctx)
-		}
+		if epochNumber%exchangeRateInterval == 0 && (epochNumber > 100) { // allow a few blocks from UpdateUndelegatedBal to avoid conflicts
+			// GET LATEST HEIGHT
+			// TODO(NOW) wrap this into a function
+			var latestHeightGaia int64 // defaults to 0
+			// get light client's latest height
+			connectionID := "connection-0"
+			conn, found := k.IBCKeeper.ConnectionKeeper.GetConnection(ctx, connectionID)
+			if !found {
+				k.Logger(ctx).Info(fmt.Sprintf("invalid connection id, \"%s\" not found", connectionID))
```

Comment:
in the above "host zone loop" we should add a "continue" here, so we don't keep processing this hostzone